yoga/Yoga.cpp

```diff
@@ -1851,7 +1851,7 @@ static bool YGNodeFixedSizeSetMeasuredDimensions(
 }
 
 static void YGZeroOutLayoutRecursivly(const YGNodeRef node) {
-  memset(&(node->getLayout()), 0, sizeof(YGLayout));
+  memset(static_cast<void*>(&(node->getLayout())), 0, sizeof(YGLayout));
```

Comment:
while you are at it, please leave the place in a better state. `static_cast<>` sidesteps the type system where we can just leverage default constructor + assignment.

```suggestion
  node->getLayout() = {};
```

Reply:
@hooddanielc, @davidaurelio  If the test results brake I think this looks like a bug of the constructor of `YGLayout` mustn't this be initialized to `{0,0}` per default in the ctor ?

Reply:
@woehrl01 That is what I thought at first, then I found some history in comments above `YGUndefined` definition.

```
/** Large positive number signifies that the property(float) is undefined.
 *Earlier we used to have YGundefined as NAN, but the downside of this is that
 *we can't use -ffast-math compiler flag as it assumes all floating-point
 *calculation involve and result into finite numbers. For more information
 *regarding -ffast-math compiler flag in clang, have a look at
 *https://clang.llvm.org/docs/UsersManual.html#cmdoption-ffast-math
 **/
#define YGUndefined 10E20F
```

I think the alternatives of using default constructor or changing the constructor may be a breaking change. Could we add a new constructor to `YGLayout`? Does the C++ delegate constructor allow us to override part of that constructors initialization list?